test/new-e2e/tests/agent-platform/step-by-step/step_by_step_test.go

```diff
@@ -0,0 +1,253 @@
+
+	fmt.Println("Parsed platform json file: ", platformJSON)
+	for _, osVers := range osVersions {
+		osVers := osVers
+		cwsSupported := false
+		for _, cwsSupportedOs := range cwsSupportedOsVersionList {
+			if cwsSupportedOs == osVers {
+				cwsSupported = true
+			}
+		}
+
+		t.Run(fmt.Sprintf("test step by step on %s %s", osVers, *architecture), func(tt *testing.T) {
+			tt.Parallel()
+			fmt.Printf("Testing %s", osVers)
+			slice := strings.Split(osVers, "-")
+			var version float64
+			if len(slice) == 2 {
+				version, err = strconv.ParseFloat(slice[1], 64)
+				require.NoError(tt, err)
+			} else if len(slice) == 3 {
+				version, err = strconv.ParseFloat(slice[1]+"."+slice[2], 64)
+				require.NoError(tt, err)
+			} else {
+				version = 0
+			}
+			majVersion, _ := strconv.Atoi(*majorVersion)
+			e2e.Run(tt, &stepByStepSuite{cwsSupported: cwsSupported, osVersion: version, agentMajorVersion: majVersion}, e2e.EC2VMStackDef(ec2params.WithImageName(platformJSON[*platform][*architecture][osVers], archMapping[*architecture], osMapping[*platform])), params.WithStackName(fmt.Sprintf("step-by-step-test-%v-%v-%s", os.Getenv("CI_PIPELINE_ID"), osVers, *architecture)))
+		})
+	}
+}
+
+func (is *stepByStepSuite) TestStepByStep() {
+	fileManager := filemanager.NewUnixFileManager(is.Env().VM)
+	unixHelper := helpers.NewUnixHelper()
+	vm := is.Env().VM.(*client.PulumiStackVM)
+	agentClient, err := client.NewAgentClient(is.T(), vm, vm.GetOS(), false)
+	require.NoError(is.T(), err)
+	VMclient := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
+
+	if *platform == "debian" || *platform == "ubuntu" {
+		is.StepByStepDebianTest(VMclient)
+	} else if *platform == "centos" || *platform == "amazonlinux" || *platform == "fedora" || *platform == "redhat" {
+		is.StepByStepRhelTest(VMclient)
+	} else {
+		require.Equal(is.T(), *platform, "suse", "NonSupportedPlatformError : %s isn't supported !", *platform)
+		is.StepByStepSuseTest(VMclient)
+	}
+	is.ConfigureAndRunAgentService(VMclient)
+	is.CheckStepByStepAgentInstallation(VMclient)
+
+}
+
+func (is *stepByStepSuite) ConfigureAndRunAgentService(VMclient *common.TestClient) {
+	is.T().Run("add config file", func(t *testing.T) {
+		ExecuteWithoutError(t, VMclient, "sudo sh -c \"sed 's/api_key:.*/api_key: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml\"")
+		ExecuteWithoutError(t, VMclient, "sudo sh -c \"chown dd-agent:dd-agent /etc/datadog-agent/datadog.yaml && chmod 640 /etc/datadog-agent/datadog.yaml\"")
+		ExecuteWithoutError(t, VMclient, "sudo systemctl restart datadog-agent.service")
+	})
+}
+
+func (is *stepByStepSuite) CheckStepByStepAgentInstallation(VMclient *common.TestClient) {
+	common.CheckInstallation(is.T(), VMclient)
+	common.CheckAgentBehaviour(is.T(), VMclient)
+	common.CheckUninstallation(is.T(), VMclient, *flavorName)
+}
+
+func (is *stepByStepSuite) StepByStepDebianTest(VMclient *common.TestClient) {
```

Comment:
I am wondering if it would be clearer to have the step by step install function in a separate file. To keep the file running the tests easier to read. WDYT?

Reply:
Since I'm using quite a few time variable like `flavorName` it's going to be hard to split into multiple files without creating circle dependencies or over-complicate the current code. And I actually think that it's still not too big where it really needs to be split. So I might just keep it this way WDYT ?